### ISSUE TITLE
can't delete notes

### ISSUE THREAD

**Issue by bestrocker221** [BoostIO/Boostnote#3270]
# Current behavior

I can't delete notes, neither with the trash icon, neither with the right click -> delete.
Notes don't get deleted

# Expected behavior

note delete

# Steps to reproduce

try to delete a note

# Environment

- Version : 5.2.11-1-MANJARO
- OS Version and name : Manjaro latest

**Comment by Flexo013** [comment]
What version of Boostnote are you using?

**Comment by bestrocker221** [comment]
The last one at the time of writing.

Carlo Alberto

**Comment by Flexo013** [comment]
Can anybody else with manjaro confirm this behavior?

**Comment by Flexo013** [comment]
@bestrocker221 Is this still an issue with 0.13.0?

**Comment by supanadit** [comment]
Same issue

**Comment by Flexo013** [comment]
@supanadit What OS are you on?

**Comment by supanadit** [comment]
@ Flexo013 5.3.11-1-MANJARO, the worst case even i can't link the other boost directory

**Comment by Flexo013** [comment]
Then I think it's safe to assume that this is a Manjaro only bug.

**Comment by kheraud** [comment]
Same problem here with last version available (0.13.0), but on arch linux `linux 5.4.2.arch1-1`

**Comment by dmouritsen** [comment]
Same issue here on Arch Linux. I can't delete notes, or folders

**Comment by supanadit** [comment]
yeah, i think we agree that boostnote can't delete any notes in any linux operating system based on arch linux such as manjaro which i used.

**Comment by TheAifam5** [comment]
I can delete on my ArchLinux.

**Comment by Flexo013** [comment]
@supanadit @dmouritsen Could you provide a gif/video of what happens on your end?

@TheAifam5 What specific version of ArchLinux are you using?

**Comment by TheAifam5** [comment]
@Flexo013 ArchLinux is rolling release distribution and does not have versions like Debian 9 or Ubuntu 18.10.

**Comment by charlwillia6** [comment]
Same here.  Cannot delete or remove from the trash.  The note stays in both the directory that I created it in and the trash.

![image](https://user-images.githubusercontent.com/9610352/76118010-fd830780-5fba-11ea-96b8-c11f3030f942.png)

**Comment by Flexo013** [comment]
If it's not the OS, then maybe it could be something problem with the path? Maybe an issue with space or other special characters.

@TheAifam5 @charlwillia6 @bestrocker221 @supanadit @dmouritsen @kheraud 
Could you post the full path to the storage you use?

**Comment by Flexo013** [comment]
Bump.

If nobody can help with further pinning down this issue, then we have no other option than to close this issue until more info is provided.

**Comment by ArmandPhilippot** [comment]
Same problem here.

> System:
  Kernel: 5.4.64-1-MANJARO x86_64 bits: 64 compiler: gcc v: 10.2.0 
  Desktop: Xfce 4.14.2 Distro: Manjaro Linux

![boostnote-bug](https://user-images.githubusercontent.com/59021693/93215962-649e2780-f767-11ea-8c52-4e5dbeb13d0c.jpg)


The same problem exists with the folders. Right clicking then "Delete folder" does not delete the folders. The only way to delete a folder is to go through "Preferences" and then click on the "Delete" button next to the folder name.

When I try to delete (whether it is a note in the trash or a folder) nothing happens. Nothing appears in the Developer Tools console either.
On the other hand, by left clicking (to select it) on a note / a folder, these messages appear in the Developer Tools console:

```
main.js:211384 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

main.js:211384 [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
```
I'm not sure it can help though ...

This issue is probably related to this https://github.com/BoostIO/Boostnote/issues/3520.

**Comment by victorbrca** [comment]
I guess there are two different issues being reported here. One is for not being able to delete notes, and another for not being able to delete the folders. I have both issues... And _I use Arch btw_

I don't get any errors on dev tools nor the terminal. I'm happy to help with more info if you need it. 

![](https://i.imgur.com/zc3c2Nz.png)

```
➤ lsb_release -a
LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
```

Installed via AUR
```
➤ pac list boostnote
local/boost-note-bin 0.9.0-1
    Boost Note.next, a replacement for Boostnote, an open source note-taking app for programmers.
    Official binary.
local/boostnote 0.16.1-1
    Open source note-taking app for programmers
```

```
➤ aur info boostnote
:: Querying AUR...
 -> Missing AUR Packages:  mplayer
Repository      : aur
Name            : boostnote
Keywords        : electron  productivity
Version         : 0.16.1-1
Description     : Open source note-taking app for programmers
URL             : https://boostnote.io/
AUR URL         : https://aur.archlinux.org/packages/boostnote
Groups          : None
Licenses        : GPL3
Provides        : None
Depends On      : electron  nodejs
Make Deps       : yarn  grunt-cli  git
Check Deps      : None
Optional Deps   : None
Conflicts With  : None
Maintainer      : clone1612
Votes           : 60
Popularity      : 0.989252
First Submitted : Mon 11 Apr 2016 09:01:00 AM EDT
Last Modified   : Sat 12 Sep 2020 04:13:16 AM EDT
Out-of-date     : No
```

**Comment by Flexo013** [comment]
Something about the file paths on ArchLinux is causing this difference. If that is handled, then both issues will probably be resolved.

**Comment by otaj** [comment]
The difference is simple. There are two packages available, that provide this app on AUR (Arch User Repository)

* https://aur.archlinux.org/packages/boostnote-bin/
  * Binary version, based on debian build, pretty much repackaged debian archive
* https://aur.archlinux.org/packages/boostnote/
  * Built from source, with couple patches (such as removing analytics, etc), using system electron

The problem only happens with the version, which is built from source. So the issue is actually probably with the building process.
Can someone from the BoostNote team please check, that the building process is correct and that the patches are not harmful to deleting functionality? Building happens in function build in [this file](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=boostnote), the patches you see mentioned in the function prepare are available [here](https://aur.archlinux.org/cgit/aur.git/tree/?h=boostnote)

**Comment by victorbrca** [comment]
I don't know if this is something that @clone1612 would be able to help with. He's the maintainer for package in AUR.

**Comment by clone1612** [comment]
I suspect it might be related to the difference in Electron version being used (as otaj mentioned the regular boostnote package uses the system version). I’ll try to look into it this weekend.

**Comment by victorbrca** [comment]
It looks like it's actually using Electron 4: 

```
victor   3222699 3222646 15 15:36 ?        00:00:02 /usr/lib/boostnote/Boostnote --type=renderer --no-sandbox --enable-features=SharedArrayBuffer --service-pipe-token=8757413579444629136 --lang=en-US --app-path=/usr/lib/boostnote/resources/app --user-agent=Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Boostnote/0.16.0 Chrome/69.0.3497.128 Electron/4.2.12 Safari/537.36 --node-integration=true --webview-tag=true --no-sandbox --background-color=#fff --enable-blink-features=OverlayScrollbars --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=8757413579444629136 --renderer-client-id=4 --shared-files=v8_context_snapshot_data:100,v8_natives_data:101
```

I tried to build electron4 on my other machine so I could compare/test but it keeps failing.

**Comment by clone1612** [comment]
I've updated the package build from source (i.e. https://aur.archlinux.org/packages/boostnote/) to now use Electron 4.

Tested delete/restore and the electron downgrade seems to resolve the issues, please update the package (latest version is ```boostnote 0.16.1-4```) and let me know if that also fixes the issues on your end.

**Comment by ArmandPhilippot** [comment]
Thanks! I just did the update and deleting notes / folders is working.

**Comment by otaj** [comment]
Can also confirm, that with electron4 it works. It would be however nice for boostnote developers to have more recent version. The reasons are simple :

1. so that we don't need to have different electron binary for different application (I currently have three electron apps, BoostNote, Ferdi and VSCode and each uses different electron package)
1. and to have bugs of older versions fixed.

**Comment by TheAifam5** [comment]
Using one electron will break a lot of things, mostly native node modules but also in some cases API.

Sorry for offtopic.
